packages/headless/src/features/insight-search/insight-search-state.ts

```diff
@@ -0,0 +1,42 @@
+import {InsightAPIErrorStatusResponse} from '../../api/service/insight/insight-api-client';
+import {
+  InsightQueryResponse,
+  SearchResult,
+} from '../../api/service/insight/query/query-response';
+
+export interface InsightSearchState {
```

Comment:
what about the other sections?
`InsightConfigurationSection &
  InsightSearchSection &
  InsightInterfaceSection &
  InsightCaseContextSection;`

Reply:
`InsightConfigurationSection`, `InsightInterfaceSection`, and `InsightCaseContextSection` are all storing information that is unique to the Insight use case, so they are not defined elsewhere. On the other hand components like the result list are expecting their data to be stored in `SearchSection`, not `InsightSearchSection`.

Reply:
Thinking about it, having an `InsightSearchState` could still work when a component accesses the state through the controller. At this point, the component does not know whether the data comes from the `search` or the `insightSearch` section, it just reads the data when updated. However, since developers might not always be using controllers, having a search response not in the `search` section might be confusing.

@nathanlb What's your take on that?

Reply:
Given we have insight search actions and reducers at this point I think it makes more sense to also define our own search state. That way we aren't bound by the current SearchState definition. 
As you said the controllers will return their own states so that's abstracted.
And my understanding is an insight panel engine would not have a `search` section in its state object if it has `insightSearch` so I don't see it being confusing.

Reply:
Just updating here that we've decided to go with the existing SearchSection and its reducers.